### ISSUE TITLE
Clippy warns `dead_code` for a function that is used

### ISSUE THREAD

**Issue by WhyNotHugo** [rust-lang/rust-clippy#10658]
### Summary

Clippy is marking a function as unused, but it's being used exactly one in the same codebase.

### Lint Name

dead_code

### Reproducer

I tried this code:

```bash
git clone https://git.sr.ht/~whynothugo/shotman
git checkout 699ddce56a4ef3c3c031f1ca522d099baa98d109
cargo clippy
```

I saw this happen:

```
warning: associated function `log_level` is never used
  --> src/cli.rs:69:19
   |
69 |     pub(crate) fn log_level(&self) -> log::Level {
   |                   ^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default
```

The function is actually used in [`main.rs:122`](https://git.sr.ht/~whynothugo/shotman/tree/699ddce56a4ef3c3c031f1ca522d099baa98d109/item/src/main.rs#L122). Using rust_analyzer's "go to definition" even takes me to the function that clippy indicates is unused.

I expected to see this happen:

nothing

### Version

```text
rustc 1.68.2 (9eb3afe9e 2023-03-27) (Alpine Linux)
binary: rustc
commit-hash: 9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0
commit-date: 2023-03-27
host: x86_64-alpine-linux-musl
release: 1.68.2
LLVM version: 15.0.7
```

**Comment by Alexendoo** [comment]
What's happening here is `cli.rs` is being compiled twice, one as a module of the lib target and once as a bin target, because there's a `mod cli;` in both `lib.rs` and `main.rs`. For the lib target the function isn't used which is where the warning is coming from

The typical solution is to have `mod cli;` in `lib.rs` only, and use it in `main.rs` and its modules via `shotman::cli` in this case

**Comment by WhyNotHugo** [comment]
If I use `use shotman::Cli` in `main.rs`, then I cannot access the method (I get an error indicating that it is private).

**Comment by Alexendoo** [comment]
Yeah, you would have to make it `pub` rather than `pub(crate)`

**Comment by WhyNotHugo** [comment]
I don't want this method to be public.
